test/test.js

```diff
@@ -78,28 +78,56 @@ describe('Routes', () => {
         retrievalId
       ])
       assert.strictEqual(rows.length, 0)
+      const result = {
+        success: true,
+        walletAddress,
+        startAt: new Date(),
+        statusCode: 200,
+        firstByteAt: new Date(),
+        endAt: new Date(),
+        byteLength: 100
+      }
       const updateRequest = await fetch(
         `${spark}/retrievals/${retrievalId}`,
         {
           method: 'PATCH',
           headers: { 'Content-Type': 'application/json' },
-          body: JSON.stringify({ success: true, walletAddress })
+          body: JSON.stringify(result)
         }
       )
       assert.strictEqual(updateRequest.status, 200)
-      const { rows: [updatedRetrievalRow] } = await client.query(`
+      const { rows: [retrievalResultRow] } = await client.query(`
         SELECT *
         FROM retrieval_results
         WHERE retrieval_id = $1
       `, [
         retrievalId
       ])
-      assert.strictEqual(updatedRetrievalRow.success, true)
-      assert.strictEqual(updatedRetrievalRow.wallet_address, walletAddress)
+      assert.strictEqual(retrievalResultRow.success, result.success)
+      assert.strictEqual(retrievalResultRow.wallet_address, walletAddress)
+      assert.strictEqual(
+        Math.round(new Date(retrievalResultRow.start_at).getTime() / 1000),
+        result.startAt.getTime()
```

Comment:
I am surprised that we are not getting Date instances from the database. Why is that? Is it because we are storing the date-time values without any timezone info?

Reply:
I was doing a bunch of mistakes while trying to get the tests to pass. Thanks for bringing it up, we should be handling timestamps correctly now

Reply:
👌🏻